File: app/src/main/java/de/bikebean/app/db/settings/settings/LocationSetting.java
package de.bikebean.app.db.settings.settings;

import androidx.annotation.NonNull;

import de.bikebean.app.db.settings.Setting;
import de.bikebean.app.db.sms.Sms;
import de.bikebean.app.db.sms.SmsFactory;
import de.bikebean.app.db.state.State;
import de.bikebean.app.db.state.StateFactory;

public abstract class LocationSetting extends Setting {

    protected LocationSetting(double location, final @NonNull Sms sms,
                              final @NonNull State.KEY key) {
        super(
                StateFactory.createStateWithEnum(
                        sms.getTimestamp() + 1, key, location, "",
                        State.STATUS.CONFIRMED, sms.getId()
                ),
                Setting::addToList
        );
    }

    protected LocationSetting(final @NonNull Sms sms, final @NonNull State.KEY key) {
        super(
                StateFactory.createNumberState(
                        sms, key, 0.0, State.STATUS.PENDING),
                Setting::addToList
        );
    }

    protected LocationSetting(final @NonNull State.KEY key) {
        super(
                StateFactory.createNumberState(
                        SmsFactory.createNullSms(), key, 0.0, State.STATUS.UNSET),
                Setting::addToList
        );
    }
}

File: app/src/main/java/de/bikebean/app/db/settings/settings/add_to_list_settings/LocationSetting.kt
package de.bikebean.app.db.settings.settings.add_to_list_settings

import de.bikebean.app.db.settings.settings.AddToListSetting
import de.bikebean.app.db.sms.Sms
import de.bikebean.app.db.state.State
import de.bikebean.app.db.state.State.KEY
import de.bikebean.app.db.state.StateFactory

abstract class LocationSetting : AddToListSetting {

    protected constructor(
            location: Double, wappState: WappState, key: KEY
    ) : super(StateFactory.createStateFromWappState(wappState, key, location))

    protected constructor(
            sms: Sms, key: KEY
    ) : super(StateFactory.createNumberState(sms, key, 0.0, State.STATUS.PENDING))

    protected constructor(
            key: KEY
    ) : super(StateFactory.createUnsetState(key, 0.0))

}